src/org/labkey/test/tests/SampleSetRemoteAPITest.java

```diff
@@ -181,6 +196,328 @@ public void sampleSetWithMissingValueField() throws IOException, CommandExceptio
+        waitAndClick(Locator.linkWithText(missingValueTable));
+        DataRegionTable materialsList =  DataRegionTable.DataRegion(getDriver()).withName("Material").waitFor();
+
+        // grab the new samples
+        SelectRowsResponse sampleSetResponse = dgen.getRowsFromServer(createDefaultConnection(true));
+        Map<String, Object> firstAddedSample = sampleSetResponse.getRows()
+                .stream()
+                .filter(a-> a.get("Name").equals("another new one"))
+                .findFirst().orElse(null);;
+        Map<String, Object> secondAddedSample = sampleSetResponse.getRows()
+                .stream()
+                .filter(a-> a.get("Name").equals("a new one"))
+                .findFirst().orElse(null);;
+
+        assertNotNull("expect new samples as outputs from batch save", firstAddedSample);
+        assertNotNull("expect new samples as outputs from batch save", secondAddedSample);
+        int secondAddedRowId = (Integer)secondAddedSample.get("RowId");
+        sleep(2000); // ugh, give it a moment
+
+        // now update one of the new rows and add another
+        String updateScript = "LABKEY.Experiment.saveBatch({\n" +
+                "  assayId: "+assayIdStringValue+",\n" +
+                "  batch: {\n" +
+                "    name: 'foo',\n" +
+                "    runs: [{\n" +
+                "      name: 'bar2',\n" +
+                "\n" +
+                "      materialOutputs: [\n" +
+                "        // create/update samples in the target sample set\n" +
+                "        { rowId: "+secondAddedRowId+", sampleSet: { name: 'mvAssaySamples' }, properties: { 'mvStringData': 'newer from api', volume: 7 } },\n" +
+                "        { name: 'another really new one', sampleSet: { name: 'mvAssaySamples' }, properties: { 'mvStringData': 'also new from api', volume:17 } }\n" +
+                "      ],\n" +
+                "\n" +
+                "      properties: {\n" +
+                "        runFileField: 'blah.png'\n" +
+                "      },\n" +
+                "\n" +
+                "      dataRows: [{\n" +
+                "        dataFileField: 'S4',\n" +
+                "        sampleId: 'def'\n" +
+                "      }]\n" +
+                "    }]\n" +
+                "  }\n" +
+                "})";
+        executeScript(updateScript);
+        sleep(2500); // ugh, again.
+
+        SelectRowsResponse updateResponse = dgen.getRowsFromServer(createDefaultConnection(true));
+        Map<String, Object> thirdAddedSample = updateResponse.getRows()
+                .stream()
+                .filter(a-> a.get("Name").equals("another really new one"))
+                .findFirst().orElse(null);
+
+        assertNotNull("expect a new sample to be added to the set", thirdAddedSample);
+        assertEquals(17.0, thirdAddedSample.get("volume"));
+        assertEquals("also new from api", thirdAddedSample.get("mvStringData"));
+
+
```

Comment:
You may want to add a TODO here as a note to check for the updated sample once that is working. 
If it turns out that update isn't valid then this part of the test adds no value (you already did a create above).

Reply:
done